File: src/forge-loom/net/auoeke/wheel/WheelForgeLoomPlugin.java
package net.auoeke.wheel;

import net.auoeke.wheel.extension.WheelForgeLoomExtension;
import net.fabricmc.loom.util.Constants;
import org.gradle.api.Project;
import org.jetbrains.annotations.NotNull;

public class WheelForgeLoomPlugin extends AbstractWheelForgePlugin<WheelForgeLoomPlugin, WheelForgeLoomExtension> implements WheelLoomPlugin<WheelForgeLoomExtension> {
    @Override
    public void apply(@NotNull Project project) {
        super.apply(project, "dev.architectury.loom", new WheelForgeLoomExtension(project));
    }

    @Override
    protected void beforeMain() {
        this.extra().set("loom.platform", "forge");
        this.extra().set("loom.forge.include", "true");
    }

    @Override
    protected void addRepositories() {
        super.addRepositories();

        this.repository("https://maven.architectury.dev");
    }

    @Override
    public void checkMinecraftVersion() {
        super.checkMinecraftVersion();
        WheelLoomPlugin.super.checkMinecraftVersion();
    }

    @Override
    public void addDependencies() {
        super.addDependencies();

        this.dependency(Constants.Configurations.MINECRAFT, "com.mojang:minecraft:" + this.extension.minecraft);
        this.dependency(Constants.Configurations.MAPPINGS, "net.fabricmc:yarn:%s+build.%s:v2".formatted(this.extension.minecraft, this.extension.yarn));
        this.dependency(Constants.Configurations.FORGE, "net.minecraftforge:forge:%s-%s".formatted(this.extension.minecraft, this.extension.forge));
    }
}

File: forge-loom/src/net/auoeke/wheel/WheelForgeLoomPlugin.kt
package net.auoeke.wheel

import net.auoeke.wheel.extension.WheelForgeLoomExtension
import net.fabricmc.loom.util.Constants
import org.gradle.api.Project

class WheelForgeLoomPlugin : AbstractWheelForgePlugin<WheelForgeLoomPlugin, WheelForgeLoomExtension>(), WheelLoomPlugin<WheelForgeLoomExtension> {
    override fun apply(project: Project) {
        super<AbstractWheelForgePlugin>.apply(project, "dev.architectury.loom", WheelForgeLoomExtension(project))
    }

    override fun beforeMain() {
        this.extra["loom.platform"] = "forge"
        this.extra["loom.forge.include"] = "true"
    }

    override fun addRepositories() {
        super.addRepositories()

        this.repository("https://maven.architectury.dev")
    }

    override fun checkMinecraftVersion() {
        super<AbstractWheelForgePlugin>.checkMinecraftVersion()
        super<WheelLoomPlugin>.checkMinecraftVersion()
    }

    override fun addDependencies() {
        super<AbstractWheelForgePlugin>.addDependencies()

        this.dependency(Constants.Configurations.MINECRAFT, "com.mojang:minecraft:${this.extension.minecraft}")
        this.dependency(Constants.Configurations.MAPPINGS, "net.fabricmc:yarn:${this.extension.minecraft}+build.${this.extension.yarn}:v2")
        this.dependency(Constants.Configurations.FORGE, "net.minecraftforge:forge:${this.extension.minecraft}-${this.extension.forge}")
    }
}
